syntaxes/cshell.tmLanguage.json

```diff
@@ -20,15 +20,19 @@
     "keywords": {
       "patterns": [
         {
-          "match": "(?<=^|;|&|\\s)(if|elif|while|until)(?=\\()",
+          "match": "(?<=^|;|&|\\s)(if|else if|switch|while)(?= *\\(.*\\))",
           "name": "keyword.control.shell"
         },
         {
-          "match": "(?<=^|;|&|\\s)(setenv|@)(?=\\s|;|&|$)",
+          "match": "(?<=^|;|&|\\s)(set|setenv|@)(?=\\s|;|&|$)",
           "name": "storage.modifier.shell"
         },
         {
-          "match": "(?<=^|;|&|\\s)(switch|end|endif|endsw|foreach)(?=\\s|;|&|$)",
+          "match": "(?<=^|;|&|\\s)(else|end|endif|then)(?=\\s|;|&|$)",
+          "name": "keyword.control.shell"
+        },
+        {
+          "match": "(?<=^|;|&|\\s)(foreach)(?= [a-zA-Z_][a-zA-Z0-9_]* *\\(.*\\))",
```

Comment:
```suggestion
```

Even though validation is a good practice when parsing programs, we can't validate syntax highlighters, because when typing the code, it should be highlighted in advance, while the user is typing it. E.g., in bash:

```shell
for
```

The `for` loops is highlighted even though it wasn't valid because the user's code should be highlighted in advance, without the syntax highlighter waiting for the user to provide arguments.

Reply:
Ok. Makes sense.